pyof/foundation/base.py

```diff
@@ -440,63 +440,82 @@ def _is_kytos_attribute(obj):
-        for _attr in self.__ordered__:
-            _class = self.__ordered__[_attr]
-            attr = getattr(self, _attr)
-            if isinstance(attr, _class):
+        for _attr, _class in self._get_attributes():
+            if isinstance(_attr, _class):
                 return True
             elif issubclass(_class, GenericType):
-                if GenericStruct._attr_fits_into_class(attr, _class):
+                if GenericStruct._attr_fits_into_class(_attr, _class):
                     return True
-            elif not isinstance(attr, _class):
+            elif not isinstance(_attr, _class):
                 return False
         return True
 
-    def get_class_attributes(self):
-        """Return a generator for class attributes' names and their types.
+    @classmethod
+    def _get_class_attributes(cls):
+        """Return a generator for class attributes' names and value.
 
         .. code-block:: python3
 
-            for _name, _type in self.get_class_attributes():
-                print("Attribute name: {}".format(_name))
-                print("Attribute type: {}".format(_type))
+            for _name, _value in self._get_class_attributes():
+                print("attribute name: {}".format(_name))
+                print("attribute type: {}".format(_value))
 
-        Returns:
-            generator: Tuples with attribute name and type.
+        returns:
+            generator: tuples with attribute name and value.
         """
-        for attribute_name in self.__ordered__:  # pylint: disable=no-member
-            yield (attribute_name, getattr(type(self), attribute_name))
+        # cls = type(self)
+        for name, value in cls.__dict__.items():
+            # gets only our (kytos) attributes. this ignores methods, dunder
+            # methods and attributes, and common python type attributes.
+            if GenericStruct._is_kytos_attribute(value):
+                yield (name, value)
 
-    def get_instance_attributes(self):
-        """Return a generator for instance attributes' names and their values.
+    def _get_instance_attributes(self):
+        """Return a generator for instance attributes' name and value.
 
         .. code-block:: python3
 
-            for _name, _value in self.get_instance_attributes():
-                print("Attribute name: {}".format(_name))
-                print("Attribute value: {}".format(_value))
+            for _name, _value in self._get_instance_attributes():
+                print("attribute name: {}".format(_name))
+                print("attribute value: {}".format(_value))
 
-        Returns:
-            generator: Tuples with attribute name and value.
+        returns:
+            generator: tuples with attribute name and value.
         """
-        for attribute_name in self.__ordered__:  # pylint: disable=no-member
-            yield (attribute_name, getattr(self, attribute_name))
+        for name, value in self.__dict__.items():
```

Comment:
Just lets leave a comment here to say that this is the new feature of python 3.6